lib/logger/lib/logger/app.ex

```diff
@@ -77,64 +65,64 @@ defmodule Logger.App do
   Stops the application without sending messages to error logger.
   """
   def stop() do
-    try do
-      Logger.Config.deleted_handlers([])
-    catch
-      :exit, {:noproc, _} ->
-        {:error, {:not_started, :logger}}
-    else
-      deleted_handlers ->
-        result = Application.stop(:logger)
-        add_handlers(deleted_handlers)
-        result
-    end
-  end
+    :ok = :logger.update_primary_config(%{level: :none})
```

Comment:
This is effectively disabling Erlang's logger, no? It is a bit dangerous because we can skip messages going to other loggers.

I think the steps here should be:

  1. Remove the Logger handler (so nothing is logged)
  2. Then stop the app

Reply:
Btw, the fact we need to do the above in steps is exactly why we had Logger.Config.deleted_handlers and we couldn't simply let Application.stop do them.

Reply:
There is also another approach, add temporary filter, that will filter out `Logger` messages, and after application stops then remove it. This will ignore only messages about the `Logger` and all other messages will be still handled.

Reply:
@hauleth I like the temporary filter and it will help keep things simple. :+1:

Reply:
I have tested this right now, and for some reason it works as expected without this line. I am not sure why this works though.

Reply:
Good call. It may be because Erlang no longer logs application stopped by default. Let's remove this line then and this should be good to go!